section3/reflection.md

```diff
@@ -1,13 +1,19 @@
 ## Section 3 Reflection
 
 1. In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?
+Felt fine.
 
 2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
+In chrome I have favorites. They probably have an array of the URLs of all the pages I've favorited.
 
 3. In your own words, explain what an Array method is.
+An Array method is a function that is defined on the Array class prototype, so that any array has access to it automatically. Some that we used this section were pop, push, shift, and unshift. The syntax for an array method is [array].method()
 
 4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
+A loop allows us to execute a single block of code multiple times. We would use a loop with an array to iterate over each member within the array. We would use a loop without an array if we already know how many times we want to do something.
 
 5. If you had to teach a 5 year old about loops, how would you explain it?
+If you have 5 apples and you want to eat each apple one at a time in a row, here's how you would do it. You'd start with the first apple, and then you'd eat that apple. Once you were done, you'd go to the next apple. You'd know you were finished once you had eaten the last apple in the line. That's what we tell a computer to do, but instead of eating apples, they're doing whatever we ask to a piece of information that belongs to a group of similar information.
 
 6. What questions do you still have about Arrays and loops?
+Are there other things we could iterate over, besides arrays? What other kinds of loops are there in Javascript, besides for loops? Is there a way to write for loops without using the triple-section syntax we used in this section? For example, Ruby has a different syntax for its for loops.
```

Comment:
The `for in` loop would be the most similar way to iterate in JS similar to how it is done in Ruby. You can see that in action towards the bottom of this [article](https://www.tutorialrepublic.com/javascript-tutorial/javascript-loops.php). You can iterate through objects as well as arrays, objects are not covered in mod 0, but definitely will be used quite a bit in the full time program.